plugin/src/main/kotlin/com/varabyte/kobweb/intellij/services/project/KobwebProjectCacheService.kt

```diff
@@ -12,36 +15,57 @@ import java.util.*
  * waste time processing code that isn't relevant to Kobweb (e.g. if we're in a multiplatform project with Kobweb,
  * Android, and other types of modules).
  *
+ * Note there's also an option to mark an element as explicitly NOT being part of a Kobweb project. This allows early
+ * aborting on followup checks, allowing the caller to distinguish the difference between "get returns null because
+ * this is not a kobweb project" vs "get returns null because we haven't put the value in the cache yet".
+ *
  * @see KobwebProject
```

Comment:
Services are a pretty standard IntelliJ concept: https://plugins.jetbrains.com/docs/intellij/plugin-services.html#retrieving-a-service so I'm not sure it makes sense to document how to  retrieve one.

See also: https://grep.app/search?q=project.service%3C

Some people add a `getInstance` method to the class's companion object, but I feel that was more a thing you had to do for Java than for Kotlin for generic type reasons -- that is, Java: `project.getService(Blah.class)` vs Kotlin: `project.service<Blah>()`

I *did* update the comment a bit, but I'm not sure that's what you were hoping for.

Otherwise, I think the API of the class itself is fairly understandable?

Reply:
Ah ok if its a standard concept then its fine, i just didnt know about it. I would prefer a companion `forProject` method on the service, but thats up to you to decide

Reply:
Based on this search: https://github.com/search?q=%22project.service%3C%22&type=code (more people seem to use raw access than wrapping in a getInstance or forProject method), I'm going to leave it. But we can revisit this pattern later if you use it for a while and continue not to like it.

Reply:
I dont really mind its fine, lets leave it at that